File: command/Door.java
package command;

public class Door {
    private String name;

    public Door(String name) {
        this.name = name;
    }

    public void lock(){
        System.out.println("Locking door...");
    }

    public void unlock() {
        System.out.println("Unlocking door...");
    }

    public String getName() {
        return name;
    }
}

File: typescript/patterns/command/door.ts
class Door {
  private name: string;

  constructor(name: string) {
    this.name = name;
  }

  lock(): void {
    console.log(`Locking Door ${this.name}...`);
  }

  unlock(): void {
    console.log(`Unlocking Door ${this.name}...`);
  }

  getName(): string {
    return this.name;
  }
}
